### PR TITLE
split off code into reconcileCommands()

### PR DESCRIPTION
**Pull request by WalterBright**
Pulls a bunch of code that manipulates `global.params` into a new function `reconcileCommands()`. Eliminates a lot of references to mutable global data.

### CODE REVIEW

**Comment by thewilsonator** [comment]
```
http://downloads.dlang.org/releases/2.x/2.079.1/dmd.2.079.1.linux.tar.xz | tar -C ../generated/host_dmd-2.079.1 -Jxf - || rm -rf ../generated/host_dmd-2.079.1
warning: failed to remove .generated/dmd-2.083.1: Directory not empty
[2018-12-10 05:35:16.448168] dautotest: Error during repository cleanup.
[2018-12-10 05:35:16.448245] dautotest: Nuking /home/dtest/DAutoTest/work/repo/dlang.org...
[2018-12-10 05:35:16.456349] dautotest: Nuking /home/dtest/DAutoTest/work/repo/.git/modules/dlang.org...
[2018-12-10 05:35:16.459181] dautotest: Updating submodule...
Cloning into '/dev/shm/dtest/work/repo/dlang.org'...
tar: ../generated/host_dmd-2.079.1: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
curl: (23) Failed writing body (2428 != 5728)
```
Dunno why though.

**Comment by WalterBright** [comment]
Looks unrelated.

File: src/dmd/mars.d [hunk 1]
```diff
@@ -2217,6 +2092,145 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
...
 }
 
+/***********************************************
+ * Adjust gathered command line switches and reconcile them.
+ * Params:
+ *      params = switches gathered from command line,
+ *               and update in place
+ *      numSrcFiles = number of source files
+ */
+private void reconcileCommands(ref Param params, size_t numSrcFiles)
```
Comment (thewilsonator) [inline src/dmd/mars.d]: This is probably useful for dmd as a library

**Review by andralex — APPROVED** [review]
Cool!

**Pull request merged**